pkg/cassandra/config/config.go

```diff
@@ -50,8 +50,8 @@ type Configuration struct {
 
 // Authenticator holds the authentication properties needed to connect to a Cassandra cluster
 type Authenticator struct {
-	Basic BasicAuthenticator `yaml:"basic" mapstructure:",squash"`
-	// TODO: add more auth types
+	Basic                 BasicAuthenticator `yaml:"basic" mapstructure:",squash"`
+	AllowedAuthenticators []string           `yaml:"allowed_authenticators" mapstructure:"allowed_authenticators"`
```

Comment:
AllowedAuthenticators should be inside BasicAuthenticator struct. And lets add a comment to it
```
// The comma-separated list of allowed password authenticators for Cassandra.
// If none are specified, there is a default "approved" list that is used
//    https://github.com/gocql/gocql/blob/34fdeebefcbf183ed7f916f931aa0586fdaa1b40/conn.go#L27
// If a non-empty list is provided, only specified authenticators are allowed.
// See also https://github.com/jaegertracing/jaeger/issues/5627#issuecomment-2174454633
```